### ISSUE TITLE
Build failure when configuring a classifier on bootRepackage task with Gradle 4.3

### ISSUE THREAD

**Issue by orange-buffalo** [spring-projects/spring-boot#10643]
When `customConfiguration` option of `bootRepackage` is used with Gradle 4.3-rc-1, build fails with an error:
```
./gradlew bootRepackage

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':bootRepackage'.
> Cannot convert the provided notation to a File or URI: task ':jar'.
  The following types/formats are supported:
    - A String or CharSequence path, for example 'src/main/java' or '/usr/include'.
    - A String or CharSequence URI, for example 'file:/usr/include'.
    - A File instance.
    - A Path instance.
    - A Directory instance.
    - A RegularFile instance.
    - A URI or URL instance.
```

Please see attached project to reproduce the problem ([spring-boot-gradle-4.3.zip](https://github.com/spring-projects/spring-boot/files/1384906/spring-boot-gradle-4.3.zip)).

This works fine with Gradle 4.2.1.

**Comment by wilkinsona** [comment]
The `customConfiguration` is a feature of Boot 1.5's Gradle Plugin which does not support Gradle 4. We'll see if we can get this to work, but you should be aware that you're using an unsupported combination here.

**Comment by orange-buffalo** [comment]
@wilkinsona Thanks a lot for the information, I was not aware about this versions incompatibility.

**Comment by wilkinsona** [comment]
The problem isn't caused by `customConfiguration` but by `classifier`. This simple script fails with Gradle 4.3 RC 1:

```groovy
plugins {
    id 'org.springframework.boot' version '1.5.7.RELEASE'
}

apply plugin: 'java'

sourceCompatibility = 1.8

repositories {
    mavenCentral()
}

bootRepackage {
    classifier 'boot'
}
```

```
$ ./gradlew bootRepackage --stacktrace

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':bootRepackage'.
> Cannot convert the provided notation to a File or URI: task ':jar'.
  The following types/formats are supported:
    - A String or CharSequence path, for example 'src/main/java' or '/usr/include'.
    - A String or CharSequence URI, for example 'file:/usr/include'.
    - A File instance.
    - A Path instance.
    - A Directory instance.
    - A RegularFile instance.
    - A URI or URL instance.

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':bootRepackage'.
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:50)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:97)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:87)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:626)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:581)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.internal.typeconversion.UnsupportedNotationException: Cannot convert the provided notation to a File or URI: task ':jar'.
The following types/formats are supported:
  - A String or CharSequence path, for example 'src/main/java' or '/usr/include'.
  - A String or CharSequence URI, for example 'file:/usr/include'.
  - A File instance.
  - A Path instance.
  - A Directory instance.
  - A RegularFile instance.
  - A URI or URL instance.
        at org.gradle.internal.typeconversion.ErrorHandlingNotationParser.parseNotation(ErrorHandlingNotationParser.java:56)
        at org.gradle.api.internal.file.AbstractFileResolver.convertObjectToFile(AbstractFileResolver.java:126)
        at org.gradle.api.internal.file.AbstractBaseDirFileResolver.doResolve(AbstractBaseDirFileResolver.java:69)
        at org.gradle.api.internal.file.AbstractFileResolver.resolve(AbstractFileResolver.java:86)
        at org.gradle.api.internal.file.AbstractFileResolver.resolve(AbstractFileResolver.java:68)
        at org.gradle.api.internal.tasks.DefaultTaskInputs.toFile(DefaultTaskInputs.java:367)
        at org.gradle.api.internal.tasks.DefaultTaskInputs.access$100(DefaultTaskInputs.java:46)
        at org.gradle.api.internal.tasks.DefaultTaskInputs$6.validate(DefaultTaskInputs.java:345)
        at org.gradle.api.internal.tasks.StaticValue.validate(StaticValue.java:43)
        at org.gradle.api.internal.tasks.DefaultTaskInputFilePropertySpec.validate(DefaultTaskInputFilePropertySpec.java:124)
        at org.gradle.api.internal.tasks.DefaultTaskInputs.validate(DefaultTaskInputs.java:161)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:47)
        ... 24 more


* Get more help at https://help.gradle.org

BUILD FAILED in 0s
3 actionable tasks: 2 executed, 1 up-to-date
```

Bizarrely, there's no Spring Boot code anywhere in the stacktrace, yet if the configuring of the classifier is removed the build passes.

Even though Boot doesn't officially support Gradle 4, given that this works with Gradle 4.2 but fails with Gradle 4.3 RC1, I think it should be reported as a regression in Gradle. I have done so [here](https://github.com/gradle/gradle/issues/3205). Hopefully the Gradle team can fix the regression before 4.3 is released and you can then continue to use Boot 1.5's Gradle plugin at your own risk. We can re-open this if they are unable to do so and explore the possibility of working around the change in behaviour.

**Comment by orange-buffalo** [comment]
@wilkinsona Thank you so much for your investigation and quick response. As far as I can see, the problem is already fixed in Gradle 4.3-rc-2, which is great news!
